### DISCUSSION TITLE
[Media Library Pro] why not define disk in config for temporary upload?

### DISCUSSION THREAD

**Discussion by Ragash** [spatie/laravel-medialibrary#2608] [Ideas] [answered]
Digging into the Pro version i noticied that temporary upload will use the default config state in:

    'disk_name' => env('MEDIA_DISK', 'public'),

if someone would like to have different behaviour for "standard media" and "temporary media" has either to

- Changing the default disk_name
- defining it's own TemporaryUpload model, containing fixed information about the disk (for example, overriding `protected static function getDiskName(): string`)

why not defining a new config, like `temp_disk_name` to be able to easy setting up a true logical "separation of concern" between tempfiles (which you should not trust) and the default ones?

thanks for your time, and keep up with the great work!

**Comment by freekmurze** [comment, marked as answer]
We haven't added this yet, because no-one asked for it, and for us it seems fine to store the temp files on the same disk.

For people that want to, they can indeed override to `TemporaryUpload` model.
